lib/Path/Tiny.pm

```diff
@@ -2188,7 +2188,7 @@ Current API available since 0.001.
 
 =cut
 
-sub stringify { $_[0]->[PATH] }
+sub stringify { (my $escaped = $_[0]->[PATH]) =~ s!^~!./~!; $escaped }
```

Comment:
I’ll see what can be done. This is a reasonably tall order. Contemporary perls do copy-on-write so the cost of simple string copies is mostly just an SV allocation and init, which will be very hard to beat, and also hard to avoid incurring the same cost in some other way. Even just doing `substr($_[0]->[PATH],0,1) eq '~'` will create an SV from the `substr()` call. Maybe this?
````
$_[0]->[PATH] =~ /^~/ ? './' . $_[0]->[PATH] : $_[0]->[PATH]
````
But that does the `$_[0][PATH]` lookup twice during runtime, and I can’t think of any way to avoid that without instead, again, allocating an SV. I guess the repeat lookup is less expensive and DaveM’s multi-lookup OP makes it cheaper still on contemporary perls.

Some Dumbbench is in order, maybe.